haystack/components/rankers/sentence_transformers_diversity.py

```diff
@@ -88,15 +150,50 @@ def __init__(
         self.device = ComponentDevice.resolve_device(device)
         self.token = token
         self.model = None
-        if similarity not in ["dot_product", "cosine"]:
-            raise ValueError(f"Similarity must be one of 'dot_product' or 'cosine', but got {similarity}.")
-        self.similarity = similarity
+        self.similarity = self.parse_similarity(similarity)
         self.query_prefix = query_prefix
         self.document_prefix = document_prefix
         self.query_suffix = query_suffix
         self.document_suffix = document_suffix
         self.meta_fields_to_embed = meta_fields_to_embed or []
         self.embedding_separator = embedding_separator
+        self.strategy = self.parse_strategy(strategy)
+        self._check_lambda_threshold(lambda_threshold, strategy)  # type: ignore
+        self.lambda_threshold = lambda_threshold or 0.5
+
+    @staticmethod
+    def parse_similarity(similarity: Union[str, Similarity]) -> "Similarity":
```

Comment:
In my opinion, this could be simplified.
In any case, I would make it internal (`_parse_similarity`), so that we are free to change it in future without breaking changes.

Same for `parse_strategy`